### ISSUE TITLE
Update raw_window_handle 0.5.0 and implement HasRawWindowHandle + HasRawDisplayHandle

### ISSUE THREAD

**Issue by EriKWDev** [PistonDevelopers/glfw-rs#513]
`raw_window_handle` has a new version which separates DisplayHandle and WindowHandle. It would be nice if glfw-rs supports these since packages like `wgpu` use it to communicate with the windowing-library in creating a surface.

Made an attempt here https://github.com/EriKWDev/glfw-rs/commit/3b3fee7d08ab53563dbaeb1d9464af4699532e11 but it doesn't quite work. Examples compile and work but I'm getting errors if I try to use it with `wgpu`.

**Comment by EriKWDev** [comment]
It seems like `winit` depends on both versions for compatability:
https://github.com/rust-windowing/winit/blob/a7a7cc64cdd13baffd71892dca445766f6e68091/Cargo.toml#L52

**Comment by nullfedora** [comment]
I just opened a pull request that fixes this problem.

**Comment by nullfedora** [comment]
https://github.com/PistonDevelopers/glfw-rs/pull/522

**Comment by coderedart** [comment]
wgpu has been using raw-window-handle `0.5` since october. I think it would be better if glfw can upgrade to that as soon as possible. `sdl2` also upgraded to `0.5` in its master branch.

**Comment by EriKWDev** [comment]
Yes. PR #524 seems like the easiest solution. Just fix the spelling error in X11 code
